script/download-grammars

```diff
@@ -150,7 +150,7 @@ end
 
 def install_grammar(tmp_dir, source, all_scopes)
   is_url = source.start_with?("http:", "https:")
-  is_single_file = source.end_with?('.tmLanguage', '.plist')
+  is_single_file = source.downcase.end_with?('.tmlanguage', '.plist') && !File.directory?(source)
```

Comment:
I think it would be better to get rid of this variable entirely and split up the two checks. Something like:

```ruby
if !is_url
  if File.directory?(source)
    DirectoryPackage.new(source)
  else
    SingleFile.new(source)
  end
elsif source.downcase.end_with?('.tmlanguage', '.plist')
  SingleGrammar.new(source)
# etc
```

That way we never call `File.directory?` on a URL, and we don't care at all about the local path's filename extension. What do you think?